index.d.ts

```diff
@@ -0,0 +1,768 @@
+    toggleFacetRefinement(...args: any[]): any;
+    toggleRefine(...args: any[]): any;
+    toggleTag(...args: any[]): any;
+    nextPage(...args: any[]): any;
+    previousPage(...args: any[]): any;
+    setCurrentPage(...args: any[]): any;
+    setPage(...args: any[]): any;
+    setQueryParameter(...args: any[]): any;
+
+    /**
+     * Set the whole state (warning: will erase previous state)
+     * @param newState the whole new state
+     * @return
+     * @fires change
+     * @chainable
+     */
+    setState(newState: QueryParameters): this;
+
+    /**
+     * Get the current search state stored in the helper. This object is immutable.
+     * @param [filters] optional filters to retrieve only a subset of the state
+     * @return if filters is specified a plain object is
+     * returned containing only the requested fields, otherwise return the unfiltered
+     * state
+     * @example
+     * // Get the complete state as stored in the helper
+     * helper.getState();
+     * @example
+     * // Get a part of the state with all the refinements on attributes and the query
+     * helper.getState(['query', 'attribute:category']);
+     */
+    getState(): SearchParameters;
+    getState(filters: string[]): QueryParameters;
+
+    getStateAsQueryString(...args: any[]): any;
+    setStateFromQueryString(...args: any[]): any;
+    overrideStateWithoutTriggeringChangeEvent(...args: any[]): any;
+    isRefined(...args: any[]): any;
+    hasRefinements(...args: any[]): any;
+    isExcluded(...args: any[]): any;
+    isDisjunctiveRefined(...args: any[]): any;
+    hasTag(...args: any[]): any;
+    isTagRefined(...args: any[]): any;
+    getIndex(...args: any[]): any;
+    getCurrentPage(...args: any[]): any;
+    getPage(...args: any[]): any;
+    getTags(...args: any[]): any;
+    getQueryParameter(...args: any[]): any;
+    getRefinements(...args: any[]): any;
+    getNumericRefinement(...args: any[]): any;
+    getHierarchicalFacetBreadcrumb(...args: any[]): any;
+    containsRefinement(...args: any[]): any;
+    clearCache(...args: any[]): any;
+    setClient(...args: any[]): any;
+    getClient(...args: any[]): any;
+    derive(...args: any[]): any;
+    detachDerivedHelper(...args: any[]): any;
+    hasPendingRequests(...args: any[]): any;
+  }
+
+  namespace AlgoliaSearchHelper {
+    /**
+     * Structure of each result when using
+     * [`searchForFacetValues()`](reference.html#AlgoliaSearchHelper#searchForFacetValues)
+     */
+    interface FacetSearchHit {
+      value: string;
+      highlighted: string;
+      count: number;
+      isRefined: boolean;
+    }
+
+    /**
+     * Structure of the data resolved by the
+     * [`searchForFacetValues()`](reference.html#AlgoliaSearchHelper#searchForFacetValues)
+     * promise.
+     */
+    interface FacetSearchResult {
+      facetHits: FacetSearchHit;
+      processingTimeMS: number;
+    }
+  }
+
+  export interface QueryParameters extends algoliasearch.QueryParameters {
```

Comment:
This interface should have a different name, right now it's confusing. When we refer to `QueryParameters` it's not clear which one it is: the one from the client or the Helper? The Helper manages its own state with a `SearchParameters` class. This class accepts a POJO that looks like class but without the methods. Inside this object every attributes are optional (but not inside `SearchParameters`).